gnovm/cmd/gno/lint.go

```diff
@@ -81,12 +85,90 @@ func execLint(cfg *lintCfg, args []string, io *commands.IO) error {
 			})
 		}
 
-		// TODO: add more checkers
+		// Use `RunMemPackage` to detect basic package errors
+		var (
+			stdout = io.Out
+			stdin  = io.In
+			stderr = io.Err
+
+			testStore = tests.TestStore(
+				rootDir, "",
+				stdin, stdout, stderr,
+				tests.ImportModeStdlibsOnly,
+			)
+
+			reParseRecover = regexp.MustCompile(`^(.+):(\d+): ?(.*)$`)
+		)
+
+		handleError := func() {
+			// Errors here mostly come from: gnovm/pkg/gnolang/preprocess.go
+			if r := recover(); r != nil {
```

Comment:
I'm not a fan of nesting; so I propose:

```suggestion
			if r := recover(); r == nil {
				return
			}
			// ... other code
```